File: typescript/src/n0104_maximum_depth_of_binary_tree.test.ts
import { expect } from "jsr:@std/expect";
import {
  maxDepth,
  maxDepthIterative,
} from "./n0104_maximum_depth_of_binary_tree.ts";
import { Tree } from "./utils/tree.ts";

Deno.test("maxDepth example one", () => {
  const tree = new Tree([3, 9, 20, null, null, 15, 7]);
  expect(maxDepth(tree.root)).toEqual(3);
});

Deno.test("maxDepth example two", () => {
  const tree = new Tree([1, null, 2]);
  expect(maxDepth(tree.root)).toEqual(2);
});

Deno.test("maxDepthIterative example one", () => {
  const tree = new Tree([3, 9, 20, null, null, 15, 7]);
  expect(maxDepthIterative(tree.root)).toEqual(3);
});

Deno.test("maxDepthIterative example two", () => {
  const tree = new Tree([1, null, 2]);
  expect(maxDepthIterative(tree.root)).toEqual(2);
});

File: rust/src/n0104_maximum_depth_of_binary_tree.test.rs
use crate::n0104_maximum_depth_of_binary_tree::*;
use crate::utils::tree::Tree;

#[test]
fn test_max_depth_example_one() {
    let tree = Tree::new([Some(3), Some(9), Some(20), None, None, Some(15), Some(7)]);
    assert_eq!(max_depth(tree.root), 3);
}

#[test]
fn test_max_depth_example_two() {
    let tree = Tree::new([Some(1), None, Some(2)]);
    assert_eq!(max_depth(tree.root), 2);
}

#[test]
fn test_max_depth_iterative_example_one() {
    let tree = Tree::new([Some(3), Some(9), Some(20), None, None, Some(15), Some(7)]);
    assert_eq!(max_depth_iterative(tree.root), 3);
}

#[test]
fn test_max_depth_iterative_example_two() {
    let tree = Tree::new([Some(1), None, Some(2)]);
    assert_eq!(max_depth_iterative(tree.root), 2);
}
